vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/TouchActions.java

```diff
@@ -150,9 +157,23 @@ public void swipeUntil(SwipeDirection direction, Duration swipeDuration, Rectang
          * - payload is {"strategy": "accessibility id", "selector": "selector"}
          * - see https://appium.io/docs/en/commands/mobile-command/#android-uiautomator2-only
          * */
-
-        int screenshotHeightToCompare = genericWebDriverManager.getSize().getHeight() / HEIGHT_DIVIDER;
-        int heightOffset = screenshotHeightToCompare * 2;
+        SearchContext searchContext = uiContext.getOptionalSearchContext().orElse(null);
+        Supplier<BufferedImage> screenShooter;
+        if (searchContext instanceof WebDriver)
+        {
+            int screenshotHeightToCompare = genericWebDriverManager.getSize().getHeight() / HEIGHT_DIVIDER;
+            int heightOffset = screenshotHeightToCompare * 2;
+            screenShooter = () -> {
+                BufferedImage bufferedImage = screenshotTaker
+                        .takeAshotScreenshot(searchContext, Optional.empty()).getImage();
```

Comment:
what if the default user strategy is full page? in previous implementation we always took viiewport only screenshot

Reply:
What full page strategy are you referring?